metal/drivers/sifive_defs.h

```diff
@@ -0,0 +1,27 @@
+/* Copyright 2020 SiFive, Inc */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#ifndef SIFIVE_DEFS_H
+#define SIFIVE_DEFS_H
+
+/*!
+ * @file sifive_defs.h
+ * @brief Common define for Sifive drivers
+ */
+
+#pragma GCC diagnostic push
+/* Accept CMSIS syntax */
+#pragma GCC diagnostic ignored "-Wreserved-id-macro"
+
+/* following defines should be used for structure members */
+#define __IM                                                                   \
```

Comment:
Any reason why these need to be in a separate file? If only used for HCA, it should live in hca header file

Reply:
I Move it into sifive_hca1_regs.h